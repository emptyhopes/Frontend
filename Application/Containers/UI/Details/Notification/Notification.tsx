import styled, { keyframes } from "styled-components";

import { useEffect } from "react";
import { IoCloseCircleOutline } from "react-icons/io5";

import { PropsInterface } from "@/Application/Containers/UI/Details/Notification/Types/PropsInterface";

const AnimationNotification = keyframes`
  0% {
    opacity: 1;
  }

  50% {
    opacity: 1;
  }

  75% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
`;

const AnimationTimer = keyframes`
  0% {
    width: 100%;
  }

  100% {
    width: 0%;
  }
`;

const StyledNotification = styled.div`
  position: absolute;
  top: 100px;
  right: 100px;
  opacity: 0;
  transition: 0.5s;

  &.active {
    animation: ${AnimationNotification} 4s linear 1;
    animation-fill-mode: forwards;
    opacity: 1;
  }
`;

const StyledContent = styled.div`
  position: relative;
  width: 400px;
  height: 50px;
  border: 1px solid #000000;
  border-radius: 10px;
  opacity: 0;
  transition: 0.5s;

  &.active {
    background-color: rgba(255, 0, 0, 0.1);
    opacity: 1;
  }
`;

const StyledTimer = styled.div`
  position: absolute;
  top: 0;
  left: 1px;
  width: 396px;
  height: 6px;
  border-radius: 10px 10px 0 0;

  &.active {
    border-radius: 10px 0 0;
    animation: ${AnimationTimer} 4s linear 1;
    animation-fill-mode: forwards;
    background-color: #ff0000;
  }
`;

const StyledButton = styled.button`
  position: absolute;
  top: 12px;
  right: 8px;
  border: none;
  background: transparent;

  & svg {
    width: 20px;
    height: 20px;
  }
`;

const StyledText = styled.p`
  position: absolute;
  top: 2px;
  left: 15px;
  color: #ff0000;
  font-size: 14px;
`;

const Notification: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  useEffect(() => {
    setTimeout(() => {
      props.SetActivated(false);
    }, 4000).unref;
  });

  const status = props.isActivated ? "active" : "";

  return (
    <>
      <StyledNotification className={status}>
        <StyledContent className={status}>
          <StyledTimer className={status} />
          <StyledButton onClick={() => props.SetActivated(false)}>
            <IoCloseCircleOutline />
          </StyledButton>
          <StyledText>{props.children}</StyledText>
        </StyledContent>
      </StyledNotification>
    </>
  );
};

export { Notification };
