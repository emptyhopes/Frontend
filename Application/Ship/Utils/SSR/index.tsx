import * as React from "react";
import * as ReactDOM from "react-dom";

import { Application } from "@/Application/Containers/Application/Application";

const Structures = <Application />;

const element = document.getElementById("root");

ReactDOM.hydrate(Structures, element);
