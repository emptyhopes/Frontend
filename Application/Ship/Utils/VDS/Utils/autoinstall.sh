#! /bin/bash

# Update and upgrade system
sudo apt-get update --yes && sudo apt-get upgrade --yes

# Remove not use packages
sudo apt-get autoremove --yes

# Install packages
sudo apt-get install git nginx certbot python3-certbot-nginx --yes

# SWAP

sudo fallocate -l 2048M /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cp /etc/fstab /etc/fstab.back

echo "/swapfile none swap sw 0 0" | sudo tee -a /etc/fstab

# NVM ( node@latest and npm@latest )

curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash

source ~/.nvm/nvm.sh

nvm install node

npm install --global npm@latest

# Install letsencrypt SSL certificate

sudo certbot
