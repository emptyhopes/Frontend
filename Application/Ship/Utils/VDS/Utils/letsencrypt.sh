#! /bin/bash

hostname="emptyhopes.ru"

sudo certbot certonly --nginx -d $hostname