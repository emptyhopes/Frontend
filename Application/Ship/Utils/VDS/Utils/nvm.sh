#! /bin/bash

temporary=$(mktemp --directory)

curl --silent --location https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh >"$temporary/install.sh"

bash "$temporary/install.sh"

source ~/.nvm/nvm.sh

nvm install node
