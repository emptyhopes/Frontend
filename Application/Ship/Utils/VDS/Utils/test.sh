#! /bin/bash

hostname="gartmann-art.ru"

curl --location -I http://$hostname.ru
curl --location -I http://en.$hostname.ru
curl --location -I http://ru.$hostname.ru

curl --location -I http://www.$hostname.ru
curl --location -I http://www.en.$hostname.ru
curl --location -I http://www.ru.$hostname.ru

curl --location -I https://$hostname.ru
curl --location -I https://en.$hostname.ru
curl --location -I https://ru.$hostname.ru

curl --location -I https://www.$hostname.ru
curl --location -I https://www.en.$hostname.ru
curl --location -I https://www.ru.$hostname.ru