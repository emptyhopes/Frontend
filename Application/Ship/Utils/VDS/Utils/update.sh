#! /bin/bash

sudo apt-get update --yes && sudo apt-get upgrade --yes
sudo apt-get autoremove --yes

sudo apt-get install git nginx certbot python3-certbot-nginx --yes