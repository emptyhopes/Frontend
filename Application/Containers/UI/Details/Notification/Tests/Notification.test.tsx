import { RenderTests } from "@/Application/Ship/Tests/Utils/RenderTests";

import { Notification } from "@/Application/Containers/UI/Details/Notification/Notification";

test("Notification", () => {
  const { container } = RenderTests(<Notification />);

  expect(container).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
