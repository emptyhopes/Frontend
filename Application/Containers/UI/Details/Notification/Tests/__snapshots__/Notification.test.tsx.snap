// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification 1`] = `
.c0 {
  position: absolute;
  top: 100px;
  right: 100px;
  opacity: 0;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c0.active {
  -webkit-animation: cKYaMk 4s linear 1;
  animation: cKYaMk 4s linear 1;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  opacity: 1;
}

.c1 {
  position: relative;
  width: 400px;
  height: 50px;
  border: 1px solid #000000;
  border-radius: 10px;
  opacity: 0;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c1.active {
  background-color: rgba(255,0,0,0.1);
  opacity: 1;
}

.c2 {
  position: absolute;
  top: 0;
  left: 1px;
  width: 396px;
  height: 6px;
  border-radius: 10px 10px 0 0;
}

.c2.active {
  border-radius: 10px 0 0;
  -webkit-animation: jMBXXb 4s linear 1;
  animation: jMBXXb 4s linear 1;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  background-color: #ff0000;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 8px;
  border: none;
  background: transparent;
}

.c3 svg {
  width: 20px;
  height: 20px;
}

.c4 {
  position: absolute;
  top: 2px;
  left: 15px;
  color: #ff0000;
  font-size: 14px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
      <button
        class="c3"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
            fill="none"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M320 320L192 192m0 128l128-128"
            fill="none"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
      </button>
      <p
        class="c4"
      />
    </div>
  </div>
</div>
`;
